src/main/java/net/mcreator/ui/modgui/ItemGUI.java

```diff
@@ -115,6 +115,17 @@
 	private final JSpinner inventorySize = new JSpinner(new SpinnerNumberModel(9, 0, 256, 1));
 	private final JSpinner inventoryStackSize = new JSpinner(new SpinnerNumberModel(64, 1, 1024, 1));
 
+	// Food parameters
+	private final JCheckBox isFood = L10N.checkbox("elementgui.common.enable");
+	private final JSpinner nutritionalValue = new JSpinner(new SpinnerNumberModel(4, -1000, 1000, 1));
+	private final JSpinner saturation = new JSpinner(new SpinnerNumberModel(0.3, -1000, 1000, 0.1));
+	private final JCheckBox forDogs = L10N.checkbox("elementgui.common.enable");
+	private final JCheckBox isAlwaysEdible = L10N.checkbox("elementgui.common.enable");
+	private ProcedureSelector onEaten;
+	private final JComboBox<String> animation = new JComboBox<>(

```

Comment:
The use animation and the "On eaten" procedure could be extended to any item, not necessarily just food ("On eaten" would be "On player finishes using item")